engine/currency_state_manager_test.go

```diff
@@ -50,7 +60,7 @@ type fakerino struct {
 
 func (f *fakerino) UpdateCurrencyStates(_ context.Context, _ asset.Item) error {
 	if f.errorMe {
-		return errors.New("norty")
+		return errExchange
```

Comment:
Can this one become `common.ErrNotYetImplemented` to utilise the error path in the function `update`? Also I really like the `errorMe` bool use for function overrides 🤙